src/se/vidstige/jadb/JadbConnection.java

```diff
@@ -25,6 +25,20 @@ public Transport createTransport() throws IOException {
         return new Transport(new Socket(host, port));
     }
 
+    public void connect(String ip, int port) throws IOException, JadbException {
+        Transport main = this.createTransport();
+        main.send("host:connect:" + ip + ":" + port);
+        main.verifyResponse();
+        main.close();
+    }
```

Comment:
Also is the first parameter really ip? It would seem it's the serial number of the device?
When you connect the device directly through wifi, it ip and port stanovjatsja its serial number (192.168.204.333:5555 example)

Let's improve the name of this method. Perhaps restart Daemon to match the adb command line utility?
I gave the name on the basis of the documentation to add

Alright, so it restarts the daemon and starts listening to the specified port. It's not described in the documentation. Have you tested this works?

Yes I checked everything works, all data changes to commit, I use the full in their work